File: pinyougou_user_web/src/main/webapp/js/service/userService.js
//服务层
app.service('userService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../user/findAll');		
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../user/findPage?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../user/findOne?id='+id);
	}
	//增加 
	this.add=function(entity,smsCode){
		return  $http.post('../user/add?smsCode='+smsCode,entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../user/update',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../user/delete?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../user/search?page='+page+"&rows="+rows, searchEntity);
	}
	this.sendSms=function(phone){
		return $http.get("../user/sendSms?phone="+phone);
	}
	this.showName=function(){
		return $http.get("../user/showName");
	}

    this.findDingdan=function () {
        return $http.get("../user/findDingdan");
    }
    this.findSeckill=function () {
        return $http.get("../user/findSeckill");
    }




    this.myInFo=function(){
		return $http.get("../user/myInFo");
	}
	this.updeteUser=function(entityUser){
		return $http.post("../user/updeteUser",entityUser);
	}
	this.address=function(){
		return $http.get("../user/address");
	}
});

File: pinyougou_user/pinyougou_user_interface/src/main/java/com/pinyougou/user/service/UserService.java
package com.pinyougou.user.service;
import com.pinyougou.common.PageResult;
import com.pinyougou.pojo.TbAddress;
import com.pinyougou.pojo.TbUser;
import com.pinyougou.pojo.vo.UserInFo;

import java.util.List;
import java.util.Map;


/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface UserService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<TbUser> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageResult findPage(int pageNum, int pageSize);
	
	
	/**
	 * 增加
	*/
	public void add(TbUser user);
	
	
	/**
	 * 修改
	 */
	public void update(TbUser user);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public TbUser findOne(Long id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Long[] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(TbUser user, int pageNum, int pageSize);

	/**
	 * 发短信的
	 */
	void sendSms(String phone);

	/**
	 * 根据手机号来校验验证码
	 * @param phone
	 * @param smsCode
	 * @return
	 */
	boolean checkSmsCode(String phone, String smsCode);

	public List<Map> findDingdan(String name);


	List<Map> findSeckill(String name);




	UserInFo myInFo(String name);

	void updateUser(UserInFo userInFo);

	List<TbAddress> address(String name);
}
